stdlib/list.mli

```diff
@@ -358,3 +366,18 @@ val to_seq : 'a list -> 'a Seq.t
 val of_seq : 'a Seq.t -> 'a list
 (** Create a list from the iterator
     @since 4.07 *)
+
+(** {1 Syntax module} *)
+module Syntax : sig
+
+  val return : 'a -> 'a list
+
+  val ( let* ) : 'a list -> ('a -> 'b list) -> 'b list
+
+  val ( and* ) : 'a list -> 'b list -> ('a * 'b) list
+
+  val ( let+ ) : 'a list -> ('a -> 'b) -> 'b list
+
+  val ( and+ ) : 'a list -> 'b list -> ('a * 'b) list
```

Comment:
For `(and+)` and `(and* )` at least you must document what operation they are performing. I think that doing it for the `let` as well would be nice, even though it's clearer from the type.